### PR TITLE
[BZ-1231203] throws if multiple mon are passed to "mon add"

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <tchaikov@gmail.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
we have py26 in tox.ini
```ini
[tox]
envlist = py26, py27, flake8
```

but seems we don't have python2.6 on the jenkins slave:
```
ERROR:   py26: InterpreterNotFound: python2.6
  py27: commands succeeded
  flake8: commands succeeded
```

**Comment by tchaikov** [comment]
@alfredodeza shall we install python2.6 on the slaves? or we should change the build matrix for this test job in jenkins?

**Comment by alfredodeza** [comment]
@tchaikov good question, I think we are OK for now, let me review this first :)

**Comment by alfredodeza** [comment]
@tchaikov this looks good. Would you mind adding a note in the help menu and the docs? That way we are even more explicit about this.

**Comment by tchaikov** [comment]
sure thing!

File: ceph_deploy/mon.py [hunk 1]
```diff
@@ -482,7 +484,8 @@ def make(parser):
 
       If the section for the monitor exists and defines a `mon addr` that
       will be used, otherwise it will fallback by resolving the hostname to an
-      IP. If `--address` is used it will override all other options.
+      IP. If `--address` is used it will override all other options. Please
+      note that only one node can be added at a time.
```
Comment (alfredodeza) [inline ceph_deploy/mon.py]: this is good. Now the only one left is in ceph-deploy/docs/sources/mon.rst

**Comment by tchaikov** [comment]
@alfredodeza thanks for your review! updated & repushed,

**Comment by trhoden** [comment]
As a note to the future, seems like a better improvement would be to remove `nargs='*'`.  No need for `nargs` at all, since `mon` is required, and we only want one host listed.  with `'*'`, it will still allow an empty list, which is not what we want.  `'+'` would at least require 1, but the check to seen that `len(args.mon) > 1` will become totally unnecessary.

**Comment by trhoden** [comment]
and if we require a list for `args.mon`, could just change to `nargs=1`.  But I see no reason for the list.

**Comment by tchaikov** [comment]
> we only want one host listed.

@trhoden probably we need to use `add_parser()` instead, because all the subcommands are sharing the same `args.mon`. for example, both `mon create` and `mon destroy` accept multiple mons. but this does not apply to `mon add`.

**Comment by alfredodeza** [comment]
Right, we are using `args.mon` in several places, all with different requirements.

The one other approach would be to make `mon_add` support multiple monitors, but that seemed like a lot more work

**Pull request merged**